webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -647,7 +648,90 @@ private static void addFmcSolutionSheet(PdfWriter docWriter, Document doc, Scram
 
             offsetTop += fontSize + marginBottom;
         }
+        
+        // the moves specs inside the table are not exactly with the rules.
+        // they are rendered separately, then rules are added above it
+        
+        // table
+        int tableHeight = 160;
+        final int tableLines = 8; // currently, we have 8 lines (3 face moves, 3 rotations and titles)
+        
+        Font smallfont = new Font(FontFamily.HELVETICA, 10);
+        PdfPTable table = new PdfPTable(8);
+        table.setTotalWidth(new float[]{25, 25, 25, 25, 25, 25, 25, competitorInfoLeft-left-7*25}); // the last part makes the moves specs fit the rules rectangle
+        table.setLockedWidth(true);
 
+        String faceMoves[] = {
+        		"F", "R", "U", "B", "L", "D"};
+        String rotations[] = {
+        		"f", "r", "u", "b", "l", "d"};
+        String directionModifiers[] = {
+        		"", "'", "2"};
+        String[] direction = {
+                translate("fmc.clockwise", locale),
+                translate("fmc.counterClockwise", locale),
+                translate("fmc.double", locale)};
+        
+        // face moves
+        PdfPCell cell = new PdfPCell(new Phrase("  - "+translate("fmc.faceMoves", locale), smallfont));
+        cell.setFixedHeight(tableHeight/tableLines);
+        cell.setColspan(8);
+        cell.setVerticalAlignment(Element.ALIGN_MIDDLE);
+        cell.setBorder(Rectangle.NO_BORDER);
+        table.addCell(cell);
+        for (int j=0; j<3; j++){
+            cell = new PdfPCell(new Phrase("", smallfont)); // offset
+            cell.setFixedHeight(tableHeight/tableLines);
+            cell.setBorder(Rectangle.NO_BORDER);
+            table.addCell(cell);
+            for (int i=0; i<6; i++){
+                cell = new PdfPCell(new Phrase(faceMoves[i]+directionModifiers[j], smallfont));
+                cell.setFixedHeight(tableHeight/tableLines);
+                cell.setVerticalAlignment(Element.ALIGN_MIDDLE);
+                cell.setHorizontalAlignment(Element.ALIGN_CENTER);
+                cell.setBorder(Rectangle.NO_BORDER);
+                table.addCell(cell);
+            }
+            cell = new PdfPCell(new Phrase(direction[j], smallfont));
+            cell.setFixedHeight(tableHeight/tableLines);
+            cell.setVerticalAlignment(Element.ALIGN_MIDDLE);
+            cell.setHorizontalAlignment(Element.ALIGN_LEFT);
+            cell.setBorder(Rectangle.NO_BORDER);
+            table.addCell(cell);
+        }
+        
+       // rotations
+        cell = new PdfPCell(new Phrase("  - "+translate("fmc.rotations", locale), smallfont));
+        cell.setFixedHeight(tableHeight/tableLines);
+        cell.setColspan(8);
+        cell.setVerticalAlignment(Element.ALIGN_MIDDLE);
+        cell.setBorder(Rectangle.NO_BORDER);
+        table.addCell(cell);
+        for (int j=0; j<3; j++){
```

Comment:
Again, can we use something more descriptive here than `3`?

Reply:
Fixed.